core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoinExec.scala

```diff
@@ -183,17 +181,45 @@ case class HashJoinExec(leftKeys: Seq[Expression],
   // return empty here as code of required variables is explicitly instantiated
   override def usedInputs: AttributeSet = AttributeSet.empty
 
+  private def findShuffleDeps(buildRDDs: Seq[RDD[_]]) : Seq[Dependency[_]] = {
+    val buildShuffleDeps: Seq[Dependency[_]] = buildRDDs.flatMap(_.dependencies
+      .filter(_.isInstanceOf[ShuffleDependency[_, _, _]]))
+    val oneToOneRdds = buildRDDs.flatMap(_.dependencies).
+      filter(_.isInstanceOf[OneToOneDependency[_]]).map(_.rdd)
+    if (oneToOneRdds.isEmpty) {
+      buildShuffleDeps
+    } else {
+      buildShuffleDeps ++ findShuffleDeps(oneToOneRdds)
+    }
+  }
+
   private lazy val (streamSideRDDs, buildSideRDDs) = {
     val streamRDDs = streamedPlan.asInstanceOf[CodegenSupport].inputRDDs()
     val buildRDDs = buildPlan.asInstanceOf[CodegenSupport].inputRDDs()
 
-    if (replicatedTableJoin) (streamRDDs, buildRDDs)
+    if (replicatedTableJoin) {
+      val streamRDD = streamRDDs.head
```

Comment:
Isn't the real problem here that "replicatedTableJoin" flag is incorrect? Are these changes required after the fix made in https://github.com/SnappyDataInc/snappydata/pull/784 that corrects the "isPartitioned" determination? Also this looks like common code in the two branches so perhaps can use it instead of repetition.